src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -390,7 +400,8 @@ public List<Setting<?>> getSettings() {
             KNN_FAISS_AVX2_DISABLED_SETTING,
             KNN_VECTOR_STREAMING_MEMORY_LIMIT_PCT_SETTING
         );
-        return Stream.concat(settings.stream(), dynamicCacheSettings.values().stream()).collect(Collectors.toList());
+        return Stream.concat(settings.stream(), Stream.concat(getFeatureFlags(), dynamicCacheSettings.values().stream()))
```

Comment:
Why not FEATURE_FLAGS.values().stream() ? and I guess we missed to make imutable?